components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java

```diff
@@ -145,6 +146,32 @@ public Response getDevices(
     }
 
     @GET
+    @Path("/user-devices")
+    public Response getDeviceByUser(@QueryParam("offset") int offset,
+                                    @QueryParam("limit") int limit) {
+
+        PaginationRequest request = new PaginationRequest(offset, limit);
+        PaginationResult result;
+        DeviceList devices = new DeviceList();
+
+        String currentUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        request.setOwner(currentUser);
+
+        try {
+            result = DeviceMgtAPIUtils.getDeviceManagementService().getDevicesOfUser(request);
+            devices.setList((List<Device>) result.getData());
+            devices.setCount(result.getRecordsTotal());
+            return Response.status(Response.Status.OK).entity(devices).build();
+        } catch (DeviceManagementException e) {
+            String msg = "Error occurred while fetching all enrolled devices";
```

Comment:
We don't need to have separate variable for Error message here "String msg",Since we uses it only one place. just put error message with in 
log.error("Error occurred while fetching all enrolled devices", e);